### DISCUSSION TITLE
Adding Structure-Block events and config options

### DISCUSSION THREAD

**Discussion by RedstoneFuture** [PaperMC/Paper#10542] [Ideas]
### Is your feature request related to a problem?

Hi. Is it possible to add events to react to the activities (saving and loading) through a `structure_block`? Could one add an option to disable these blocks?

After all these years, I wonder why there are no settings / interfaces here for performant checks? On a creative server, the structure-block can theoretically generate restriction problems. Normally, no structure-block can be placed with NBT values, but it would be a bit safer if there were a few more possibilities through Vanilla/SpigotMC/PaperMC.

### Describe the solution you'd like.

For example, a StructureBlockSave / Load Event or a config option to disable the activities generally.

### Describe alternatives you've considered.

- Cancel of `onBlockRedstoneChange` event if the target block is a `structure_block` (not efficient)
- Deactivation of OP (already existing blocks with NBT content can be further activated with Redstone)

**Comment by electronicboy** [comment]
Structures have never really been exposed in the API afaik, there's just been virtually no demand for them given that
1) There is no NBT access in the API
2) There have been much better long standing alternatives vs structures within the community

**Comment by RedstoneFuture** [comment]
I never use the blocks.
It's a security issue for me that they can't be 100% disabled.

**Comment by Machine-Maker** [comment]
I think you can use the packet limiter to just drop all packets of that type (`ServerboundSetStructureBlockPacket`). It's possible the packet limiter doesn't accept 0, I don't remember.

**Comment by Phoenix616** [comment]
Any chance that we will at least get an event for block changes? Right now there isn't really any way to catch blocks getting placed by structures :S
